### ISSUE TITLE
ClassCastException with Slick

### ISSUE THREAD

**Issue by osbornk** [opentracing-contrib/java-specialagent#357]
I am using OpenTracing Special Agent with Play and Slick. If I include the "concurrent" rule, I get the following error with Slick.

```
[error] ErrorHandler - class io.opentracing.contrib.specialagent.rule.concurrent.TracedRunnable cannot be cast to class slick.util.AsyncExecutor$PrioritizedRunnable (io.opentracing.contrib.specialagent.rule.concurrent.TracedRunnable is in unnamed module of loader 'bootstrap'; slick.util.AsyncExecutor$PrioritizedRunnable is in unnamed module of loader play.runsupport.NamedURLClassLoader @70355caa)
java.lang.ClassCastException: class io.opentracing.contrib.specialagent.rule.concurrent.TracedRunnable cannot be cast to class slick.util.AsyncExecutor$PrioritizedRunnable (io.opentracing.contrib.specialagent.rule.concurrent.TracedRunnable is in unnamed module of loader 'bootstrap'; slick.util.AsyncExecutor$PrioritizedRunnable is in unnamed module of loader play.runsupport.NamedURLClassLoader @70355caa)
	at slick.util.ManagedArrayBlockingQueue.offer(ManagedArrayBlockingQueue.scala:17)
	at java.base/java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1347)
	at slick.util.AsyncExecutor$$anon$1$$anon$4.execute(AsyncExecutor.scala:163)
	at slick.basic.BasicBackend$DatabaseDef.runSynchronousDatabaseAction(BasicBackend.scala:265)
	at slick.basic.BasicBackend$DatabaseDef.runSynchronousDatabaseAction$(BasicBackend.scala:263)
```

Looks like it is trying to instrument something it shouldn't. If there is any further information I can provide, please let me know.

If I don't include the concurrent plugin, there are no errors. But then the context is not propagated properly.

**Comment by safris** [comment]
Hi @osbornk, thank you for this issue. Coincidentally, I was researching the underlying cause of this issue that was related to similar situations with other plugins. This is a critical issue, and I have checked in the fix for it in #359.

I have pushed the release date of [v1.5.6](https://github.com/opentracing-contrib/java-specialagent/milestone/3) to today. Once [this release build](https://travis-ci.org/opentracing-contrib/java-specialagent/builds/638708645) is finished, v1.5.6 will be available on Maven Central.
